### DISCUSSION TITLE
Full Changelog?

### DISCUSSION THREAD

**Discussion by mirkonasato** [vercel/next.js#65212] [Help]
### Summary

Hi all,

Is there a place where one can easily see the full changelog for all (and only) stable releases, like e.g. [react/CHANGELOG.md](https://github.com/facebook/react/blob/main/CHANGELOG.md)? Whenever I need to find out in which Next.js version something changed it's a long and time consuming investigation.

[github.com/vercel/next.js/releases](https://github.com/vercel/next.js/releases) includes all `canary` releases, which makes it difficult to see only the stable ones, and each release is a separate page anyway. Announcements at [nextjs.org/blog](https://nextjs.org/blog) mostly show new features at a high level, but not the full details.

Ideally I'd like a single page with all the changes made in all 14.x releases, then another for v13.x etc. If there isn't such a resource already I may try putting it together.

Thanks

---

Related but not quite the same discussions:
* [Where can I find a changelog for human?](https://github.com/vercel/next.js/discussions/59942)
* [Better changelogs](https://github.com/vercel/next.js/discussions/58405)

**Comment by petrvecera** [comment]
I second this option, trying to find changelog for 14.2.7 and I am baffled that I can't find it under few minutes. Actually I haven't find it yet. 🤯

**Reply by Xavier-IV** [reply]
> I second this option, trying to find changelog for 14.2.7 and I am baffled that I can't find it under few minutes. Actually I haven't find it yet. 🤯

It's weird that Twitter lead me to this releases (from googling) instead of official resources.

Here's the full list, just change the URL to the releases that you'd like, except that this are for each patch releases.

https://github.com/vercel/next.js/releases/tag/v14.2.7
https://github.com/vercel/next.js/releases/tag/v14.2.6
https://github.com/vercel/next.js/releases/tag/v14.2.5
https://github.com/vercel/next.js/releases/tag/v14.2.4
https://github.com/vercel/next.js/releases/tag/v14.2.3
https://github.com/vercel/next.js/releases/tag/v14.2.2

But I agree that this should be easily discoverable, especially for big releases.

**Reply by trevorh** [reply]
@leerob Is there a way to get this on nextjs.org so it's more discoverable? ~GH doesn't provide an easy way to filter the releases by semvar.~ Thanks!

PS I was wrong; there is a releases filter box so I can enter v14, but it would be nice if there was a more official changelog closer to the docs.

**Reply by valentijnscholten** [reply]
There used to be this: https://next-changelog.vercel.app/ which is no longer updated.

**Reply by jvandenaardweg** [reply]
Here you go: https://nextjs-changelog.vercel.app/?version=14.2.7:latest

**Reply by arsinclair** [reply]
@jvandenaardweg genius, thank you! In case you're the author of the project: it would be cool to convert `#1234` links to the actual links to the relevant PRs. Github is doing that automatically, but this cute changelog website doesn't have that context.

**Reply by malezjaa** [reply]
Hey, i am the author of https://next-changelog.vercel.app/ which should be working now.

**Comment by jvandenaardweg** [comment]
Just needed something similar, so created a page myself: https://nextjs-changelog.vercel.app/ (bookmark it!)

Filtered out all the canary and rc crap. Only stable release notes.

Page is a little heavy since it's rendering all versions with all changelog items so you can just CTRL + F. Will optimize that later.

Also allows for only showing the relevant release notes for your upgrade path, say like from `14.2.7` to `latest`: 
https://nextjs-changelog.vercel.app/?version=14.2.7:latest

Updates automatically on new releases

**Comment by Netail** [comment]
Like this? https://github.com/vercel/next.js/releases?q=prerelease%3Afalse&expanded=true

GitHub offers some search filters; https://docs.github.com/en/repositories/releasing-projects-on-github/searching-a-repositorys-releases#search-syntax-for-searching-releases-in-a-repository
